devices/hip/executor.cpp

```diff
@@ -61,10 +61,38 @@ bool HipExecutor::verify_memory_to(const CudaExecutor *dest_exec) const
 }
 
 
-int HipExecutor::num_execs[max_devices];
+#if (GINKGO_HIP_PLATFORM_NVCC == 1)
+using hip_device_class = NvidiaDevice;
+#else
+using hip_device_class = AmdDevice;
+#endif
+
+
+void HipExecutor::increase_num_execs(int device_id)
+{
+#ifdef GINKGO_BUILD_HIP
+    // increase the HIP Device count only when ginkgo build hip
+    std::lock_guard<std::mutex> guard(hip_device_class::get_mutex(device_id));
+    hip_device_class::get_num_execs(device_id)++;
```

Comment:
Just curious, why `snake_case` here and `CamelCase` for `NvidiaDevice` and `AmdDevice` ?

Reply:
NvidiaDevice and AmdDevice are CamelCase because they are class.
I was thinking hip_device_class is like `using value_type = ValueType`, so I put it snake_case.
Is there any rule we can follow?

Reply:
Technically, the policy is the following: https://github.com/ginkgo-project/ginkgo/wiki/Contributing-guidelines#structures-and-classes

i.e., snake_case for all nonpolymorphic classes, CamelCase for polymorphic behavior. I don't think we are very consistent on that topic though.